package.nix

```diff
@@ -0,0 +1,50 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, installShellFiles
+}:
+
+let
+  cargoToml = lib.importTOML ./Cargo.toml;
+in
+
+rustPlatform.buildRustPackage rec {
+  pname = "ast-grep";
+  version = cargoToml.workspace.package.version;
+
+  src = ./.;
```

Comment:
That would be needed if we were in another repo, like [nixpkgs](https://github.com/nixos/nixpkgs). But since we're in the `ast-grep` repository, we need to include the code that's under source control here.

This could be better: as written, if any of the nix files change, the whole thing rebuilds. I think that's a fine starting place, though.